### DISCUSSION TITLE
How do I get a JSON list/array using Ktorfit?

### DISCUSSION THREAD

**Discussion by LionZXY** [Foso/Ktorfit#325] [Q&A] [answered]
## Background

I have a some object, for example:
```kotlin
data class Person(val name: String)
```
And some API that returns an array of these Person:
```json
[{"name": "John"}, {"name": "Alex"}]
```

## Question

When I use the pure ktor api I can use this API like this:
```kotlin
val response = client.get("https://persons.example/getAll").body<List<Person>>()
println(response)
```
Result: `[Person(name=John), Person(name=Alex)]`

When i use retrofit i can use this API like this:
```kotlin
interface PersonApi {
    @GET("getAll")
    suspend fun getPersons(): List<Person>
}

val response = personApi.getPersons()
println(response)
```
Result: `[Person(name=John), Person(name=Alex)]`

BUT
When i try use ktorfit with same interface like retrofit, with same ktor HttpClient, i receive this error:
```plaintext
Caused by: kotlinx.serialization.json.internal.JsonDecodingException: Expected start of the object '{', but had 'EOF' instead at path: $
```

How I can get a JSON list/array using Ktorfit?

**Comment by Foso** [comment]
Hi, it should work that way. Are you setting the right baseUrl? Are you getting a response with:

`suspend fun getPersons(): String`  ?

**Reply by LionZXY** [reply]
Yes
Yes, you can try adding a method in your example that returns an array and it won't work either

**Reply by LionZXY** [reply]
You can try running the android example from here:
https://github.com/LionZXY/Ktorfit/tree/array_application
And look at this code:
https://github.com/LionZXY/Ktorfit/blob/array_application/example/AndroidOnlyExample/app/src/main/java/de/jensklingenberg/androidonlyexample/MainActivity.kt#L64-L71
```kotlin
// Loaded successful:
val response = api.getApplicationsString()
Log.i("MainActivity", response)
// Decoded successful:
val decodedAnswer = json.decodeFromString<Array<KtorfitApplicationShort>>(response)
Log.i("MainActivity", decodedAnswer.toList().toString())
// Crash:
peopleState.value = api.getApplications().firstOrNull()
```

**Comment by Foso** [comment, marked as answer]
Thank you for the example project, i can reproduce the issue, it`s a bug. A workaround is to wrap it in a Response `Response<Array<KtorfitApplicationShort>>` . I`m planning to fix the bug today

**Comment by Foso** [comment]
Ktorfit 1.4.1 is now released
